yabt/caching.py

```diff
@@ -443,6 +443,7 @@ def try_use_global_cache(build_context, func, error_msg):
             return func()
         except Exception as e:
```

Comment:
I'm guessing it will [google.api_core.exceptions.TooManyRequests](https://googleapis.dev/python/google-api-core/latest/exceptions.html#google.api_core.exceptions.TooManyRequests).
So I would assume that and add the log you added so we can fix it in case I'm guessing wrong.